src/app/core/params.ts

```diff
@@ -3,27 +3,22 @@ import Parameter from "./parameter";
  * Params class
  */
 export default class Params {
+    private unique: boolean;
+    private heritable: boolean;
+    parameters: any;
```

Comment:
This shouldn't be any is. Should be a list/map of `Parmeter`

Reply:
Is parameters equivalent to the values argument in the constructor? Otherwise parameters is never given values.